apps/dropbox/src/connectors/utils/format-shared-links.ts

```diff
@@ -0,0 +1,30 @@
+import { sharing } from 'dropbox';
+import { SharedLinks } from '../types';
+
+export const filterSharedLinks = (sharedLinks: sharing.ListSharedLinksResult['links']) => {
+  return sharedLinks.reduce((acc: SharedLinks[], link) => {
+    const { id, url, link_permissions, path_lower } = link;
+
+    const effectiveAudience =
+      link_permissions.effective_audience?.['.tag'] ??
+      link_permissions.resolved_visibility?.['.tag'] ??
+      null;
+
+    if (!id || !effectiveAudience || !link?.path_lower) {
+      return acc;
+    }
+
+    if (!['password', 'public'].includes(effectiveAudience)) {
+      return acc;
+    }
+
+    const linkObject = {
+      id,
+      url,
+      linkAccessLevel: link.link_permissions.link_access_level?.['.tag'] ?? 'viewer',
+      pathLower: path_lower!,
+    };
+
+    return [...acc, linkObject];
```

Comment:
```suggestion
    acc.push(linkObject)
    return acc;
```
https://www.measurethat.net/Benchmarks/Show/29740/0/reduce-array-destructuring-vs-push